.github/workflows/main.yml

```diff
@@ -903,6 +893,24 @@ jobs:
 #      - name: java/ant.grammar
 #        run: ant $OPTS -f java/ant.grammar test
 
+      - name: Set up JDK 17 for JDK 21 incompatible tests
+        if: ${{ matrix.java == '21' }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: ${{ env.default_java_distribution }}
+
+      # TODO fix JDK 21 incompatibilites
+      - name: java/java.mx.project
+        continue-on-error: ${{ github.event_name != 'pull_request' }}
+        run: .github/retry.sh ant $OPTS -f java/java.mx.project test
+
+      - name: java/gradle.java
+        run: .github/retry.sh ant $OPTS -f java/gradle.java test
+
+      - name: extide/gradle
+        run: ant $OPTS -f extide/gradle test
+
       - name: extide/o.apache.tools.ant.module
         run: ant $OPTS -f extide/o.apache.tools.ant.module test
```

Comment:
gradle should fix itself automatically once it is actually compatible with JDK 21

`java.mx.project` and `o.apache.tools.ant.module` failed on JDK 21